src/dsp_tools/utils/xml_validation_models.py

```diff
@@ -0,0 +1,143 @@
+from dataclasses import dataclass
+
+import pandas as pd
+from lxml import etree
+
+list_separator = "\n    - "
+medium_separator = "\n----------------------------\n"
+grand_separator = "\n\n---------------------------------------\n\n"
+
+maximum_prints = 50
+
+
+@dataclass
+class TextValueData:
+    resource_id: str
+    property_name: str
+    encoding: set[str]
+
+
+def check_if_only_one_encoding_is_used_in_xml(
+    root: etree._Element,
+) -> tuple[bool, list[TextValueData] | None]:
+    """
+    This function analyses if all the encodings in the <text> elements are consistent within one <text-prop>
+
+    This is correct:
+    ```
+    <text-prop name=":hasSimpleText">
+        <text permissions="prop-default" encoding="utf8">Text 1</text>
+        <text permissions="prop-default" encoding="utf8">Text 2</text>
+    </text-prop>
+    ```
+
+    This is wrong:
+    ```
+    <text-prop name=":hasSimpleText">
+        <text permissions="prop-default" encoding="utf8">Text 1</text>
+        <text permissions="prop-default" encoding="xml">Text 2</text>
+    </text-prop>
+    ```
+
+    Args:
+        root: root of the data xml document
+
+    Returns:
+          True and None if all the elements are consistent
+          False and a list of all the inconsistent <text-props>
+    """
+    text_props = _get_all_ids_prop_encoding_from_root(root)
+    if (checked_props := _check_only_one_valid_encoding_used_all_props(text_props)) is not None:
```

Comment:
Is it possible that you encountered this with pandas dataframe comparison? I'm almost 100% sure that in pure python, my suggestion is cleaner and will never raise a future warning. I don't think that your variant is a good habit. (Except for pandas, there it might be necessary)

Reply:
In lxml you may also get these warnings, I think. But it's true that in pure python this behaviour guaranteed to work - the only case where you need `is not None` is when you want to distinguish actual `None` from some falsy value like empty string, empty list, `False`, etc.